language-extensions/dotnet-core-CSharp/src/managed/CSharpInputDataSet.cs

```diff
@@ -129,7 +129,25 @@ private unsafe void AddColumn(
                 case SqlDataType.DotNetChar:
                     int[] strLens = new int[rowsNumber];
                     Interop.Copy((int*)colMap, strLens, 0, (int)rowsNumber);
-                    CSharpDataFrame.Columns.Add(new StringDataFrameColumn(_columns[columnNumber].Name, DataSetUtils.StringSplitToArray(Interop.UTF8PtrToStr((char*)colData), strLens)));
+
+                    // Use byte-based splitting to properly handle multi-byte UTF-8 characters
```

Comment:
can you please explain this?
It also looks you're justifying the code we're replacing while in future that code does not exist anymore (so, the reader would think byte-based in compared to?)

Let's say what do we do here and why.